### PR TITLE
Application authorization checking, resetting and revoking

### PR DESCRIPTION
**Pull request by alfhenrik**
Add functionality for checking, resetting and revoking OAuth application tokens

- [x] Add tests
- [x] Add integration tests
- [x] @shiftkey is happy

Fixes https://github.com/octokit/octokit.net/issues/682
Fixes https://github.com/octokit/octokit.net/issues/652
Fixes https://github.com/octokit/octokit.net/issues/653
Fixes https://github.com/octokit/octokit.net/issues/654

### CODE REVIEW

**Comment by alfhenrik** [comment]
This should now be :ok_hand: for review, as long as the AppVeyour build doesn't go :boom: :wink:

File: Octokit.Reactive/Clients/IObservableAuthorizationsClient.cs [hunk 1]
```diff
@@ -78,6 +78,52 @@ public interface IObservableAuthorizationsClient
             string twoFactorAuthenticationCode);
 
         /// <summary>
+        /// </summary>
```
Comment (shiftkey) [inline Octokit.Reactive/Clients/IObservableAuthorizationsClient.cs]: :memo: a summary for this method?

File: Octokit.Tests.Integration/Clients/AuthorizationClientTests.cs [hunk 2]
```diff
@@ -97,5 +98,111 @@ public class AuthorizationClientTests
...
+                Helper.ClientSecret,
+                newAuthorization);
+
+            var applicationClient = Helper.GetAuthenticatedApplicationClient();
+            var applicationAuthorization = await applicationClient.Authorization.CheckApplicationAuthentication(Helper.ClientId, created.Token);
+
+            Assert.NotNull(applicationAuthorization);
+            Assert.Equal(created.Token, applicationAuthorization.Token);
+
+            await client.Authorization.Delete(created.Id);
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/AuthorizationClientTests.cs]: Is it worth asserting the action succeeded here?

File: Octokit/Clients/IAuthorizationsClient.cs [hunk 3]
```diff
@@ -100,6 +100,52 @@ public interface IAuthorizationsClient
             string twoFactorAuthenticationCode);
 
         /// <summary>
+        /// </summary>
```
Comment (shiftkey) [inline Octokit/Clients/IAuthorizationsClient.cs]: :memo: a summary here for the method?

**Comment by shiftkey** [comment]
Looks good, just a couple of documentation-related things.

I'll kick the tires on the tests today and see what shakes out :v:

File: Octokit/Clients/IAuthorizationsClient.cs (hunk 3 above)
Comment (alfhenrik) [inline Octokit/Clients/IAuthorizationsClient.cs]: good :eyes:!

**Comment by alfhenrik** [comment]
Dun! :triumph: 

**Side Note:** I was able to get around the Integration Tests time out issue, by just running my Integration Tests from VS.

**Comment by shiftkey** [comment]
> I was able to get around the Integration Tests time out issue, by just running my Integration Tests from VS.

Yeah, that's what I do too (because it doesn't enforce the timeout :wink:)

**Comment by shiftkey** [comment]
All looks pretty great to me. Thanks for finishing off this feature!

**Pull request merged**